mlir/include/Quantum/Transforms/Passes.td

```diff
@@ -88,4 +82,23 @@ def AnnotateFunctionPass : Pass<"annotate-function"> {
     let constructor = "catalyst::createAnnotateFunctionPass()";
 }
 
+// ----- Quantum circuit transformation passes begin ----- //
+// For example, automatic compiler peephole opts, etc.
+
+class QuantumCircuitTransformationPass<string CmdlineName> : Pass<CmdlineName> {
+    let options = [
+    Option<"FuncNameOpt", "func-name",
+           "std::string", /*default=*/"\"\"",
+           "The name of the function (or qnode) to run the quantum circuit transformation pass on.">,
+    ];
```

Comment:
Something that I think should be explored is adding a tag to a particular function and then just running this pass on all functions on that have that pass. This avoids the need of having to know the name of functions and having multiple different passes one for each function.

Reply:
Interesting thought, so instead of using another pass table, we just tag the function (qnode) object itself, and generate pipeline strings from the tags? This would eliminate the use of the current global `active_passes` table and would be a significant improvement!

Reply:
Yes. I think we already do that to some extent when looking for the qnode attribute during the asynchronous compilation. Until we have a proper abstraction of a "qnode module" and then encode the passes we want directly on the IR, targetting a specific module, I think this is good enough. And since all quantum operations are always inlined in a single function this will work. Once this invariant is broken (but before we have the qnode module abstraction), we would need to extend this pass such that all functions reached by the qnode entry point should also run cancel inverses.

Reply:
Resolving this as we decided to use the transform dialect to schedule the passes